nautobot/extras/tests/test_filters.py

```diff
@@ -958,6 +960,189 @@ def test_destination_id(self):
+            VLAN.objects.create(vid=2, name="VLAN 2"),
+            VLAN.objects.create(vid=3, name="VLAN 3"),
+        )
+        cls.relationship_associations = (
+            RelationshipAssociation(
+                relationship=cls.relationships[0],
+                source_type=cls.device_type,
+                source_id=cls.devices[0].pk,
+                destination_type=cls.vlan_type,
+                destination_id=cls.vlans[0].pk,
+            ),
+            RelationshipAssociation(
+                relationship=cls.relationships[0],
+                source_type=cls.device_type,
+                source_id=cls.devices[1].pk,
+                destination_type=cls.vlan_type,
+                destination_id=cls.vlans[1].pk,
+            ),
+            RelationshipAssociation(
+                relationship=cls.relationships[1],
+                source_type=cls.vlan_type,
+                source_id=cls.vlans[0].pk,
+                destination_type=cls.device_type,
+                destination_id=cls.devices[0].pk,
+            ),
+            RelationshipAssociation(
+                relationship=cls.relationships[1],
+                source_type=cls.vlan_type,
+                source_id=cls.vlans[0].pk,
+                destination_type=cls.device_type,
+                destination_id=cls.devices[1].pk,
+            ),
+            RelationshipAssociation(
+                relationship=cls.relationships[1],
+                source_type=cls.vlan_type,
+                source_id=cls.vlans[0].pk,
+                destination_type=cls.device_type,
+                destination_id=cls.devices[2].pk,
+            ),
+            RelationshipAssociation(
+                relationship=cls.relationships[2],
+                source_type=cls.device_type,
+                source_id=cls.devices[0].pk,
+                destination_type=cls.device_type,
+                destination_id=cls.devices[1].pk,
+            ),
+            RelationshipAssociation(
+                relationship=cls.relationships[2],
+                source_type=cls.device_type,
+                source_id=cls.devices[0].pk,
+                destination_type=cls.device_type,
+                destination_id=cls.devices[2].pk,
+            ),
+            RelationshipAssociation(
+                relationship=cls.relationships[2],
+                source_type=cls.device_type,
+                source_id=cls.devices[1].pk,
+                destination_type=cls.device_type,
+                destination_id=cls.devices[2].pk,
+            ),
+        )
+        for relationship_association in cls.relationship_associations:
+            relationship_association.validated_save()
+
+    def test_one_to_many_source(self):
+        self.queryset = Device.objects.all()
+        self.filterset = DeviceFilterSet
+        self.assertEqual(
+            self.filterset({f"cr_{self.relationships[1].slug}__source": self.vlans[0].pk}, self.queryset).qs.count(), 3
```

Comment:
Even though this is a one-to-many relationship, the *filter* should still accept a list of pks, rather than a single value.